src/language-js/printer-estree.js

```diff
@@ -2876,7 +2895,27 @@ function printPathNoParens(path, options, print, args) {
     }
 
     case "TypeParameterDeclaration":
-    case "TypeParameterInstantiation":
+    case "TypeParameterInstantiation": {
+      const value = path.getValue();
+      const commentStart = value.range
+        ? options.originalText.substring(0, value.range[0]).lastIndexOf("/*")
+        : -1;
+      // As noted in the TypeCastExpression comments above, we're able to use a normal whitespace regex here
+      // because we know for sure that this is a type definition.
+      const commentSyntax =
+        commentStart >= 0 &&
+        options.originalText.substring(commentStart).match(/^\/\*\s*::/);
```

Comment:
Are newlines allowed between `/*` and `::`? I'm thinking of https://github.com/facebook/flow/issues/7050. Maybe something similar applies here?

Reply:
Doesn't seem like it: https://flow.org/try/#0PQKgUAXBAEAuCeAHAptAYgew9AvNAzrAE4CWAdgOYDcYIwYYAxhmYdAGZagybZ27QA5IKpA

That said, we should be safe to use `\s*` as our whitespace match here since we know we're looking at a `TypeParameterDeclaration` or `TypeParameterInstantiation` AST node.